src/Storage/Storage.Management.Test/ScenarioTests/StorageAccountTests.ps1

```diff
@@ -1770,67 +1770,69 @@ function Test-StorageBlobInventory
+        Assert-AreEqual 3 $policy1.Rules.Count
+
+        # get inventory policy 
+        $policy1 = Get-AzStorageBlobInventoryPolicy -ResourceGroupName $rgname  -StorageAccountName $stoname
+        Assert-AreEqual $false $policy1.Enabled
+        Assert-AreEqual 3 $policy1.Rules.Count
+        Assert-AreEqual "test1" $policy1.Rules[0].Name
+        Assert-AreEqual $false $policy1.Rules[0].Enabled
+        Assert-AreEqual $containerName $policy1.Rules[0].Destination
+        Assert-AreEqual "Parquet" $policy1.Rules[0].Definition.Format
+        Assert-AreEqual "Weekly" $policy1.Rules[0].Definition.Schedule
+        Assert-AreEqual "Blob" $policy1.Rules[0].Definition.ObjectType
+        Assert-AreEqual 14 $policy1.Rules[0].Definition.SchemaFields.Count
+        Assert-AreEqual $true $policy1.Rules[0].Definition.Filters.IncludeSnapshots
+        Assert-AreEqual $true $policy1.Rules[0].Definition.Filters.IncludeBlobVersions
+        Assert-AreEqual 2 $policy1.Rules[0].Definition.Filters.BlobTypes.Count
+        Assert-AreEqual 5 $policy1.Rules[0].Definition.Filters.PrefixMatch.Count
+        Assert-AreEqual "test2" $policy1.Rules[1].Name
+        Assert-AreEqual $false $policy1.Rules[1].Enabled
+        Assert-AreEqual $containerName $policy1.Rules[1].Destination
+        Assert-AreEqual "Csv" $policy1.Rules[1].Definition.Format
+        Assert-AreEqual "Daily" $policy1.Rules[1].Definition.Schedule
+        Assert-AreEqual "Container" $policy1.Rules[1].Definition.ObjectType
+        Assert-AreEqual 12 $policy1.Rules[1].Definition.SchemaFields.Count
+        Assert-Null $policy1.Rules[1].Definition.Filters.IncludeSnapshots
+        Assert-Null $policy1.Rules[1].Definition.Filters.IncludeBlobVersions
+        Assert-AreEqual 2 $policy1.Rules[1].Definition.Filters.PrefixMatch.Count
+        Assert-AreEqual "test3" $policy1.Rules[2].Name
+        Assert-AreEqual $true $policy1.Rules[2].Enabled
+        Assert-AreEqual $containerName $policy1.Rules[2].Destination
+        Assert-AreEqual "Csv" $policy1.Rules[2].Definition.Format
+        Assert-AreEqual "Weekly" $policy1.Rules[2].Definition.Schedule
+        Assert-AreEqual "Blob" $policy1.Rules[2].Definition.ObjectType
+        Assert-AreEqual 7 $policy1.Rules[2].Definition.SchemaFields.Count
+        Assert-Null $policy1.Rules[2].Definition.Filters.IncludeSnapshots
+        Assert-Null $policy1.Rules[2].Definition.Filters.IncludeBlobVersions
+        Assert-AreEqual $true $policy1.Rules[2].Definition.Filters.IncludeDeleted
+        Assert-AreEqual 2 $policy1.Rules[2].Definition.Filters.ExcludePrefix.Count
+        Assert-AreEqual 1 $policy1.Rules[2].Definition.Filters.BlobTypes.Count
+        Assert-AreEqual 2 $policy1.Rules[2].Definition.Filters.PrefixMatch.Count
+
+        # set policy with json and account name pipeline
+        $policy2 = Get-AzStorageAccount -ResourceGroupName $rgname  -StorageAccountName $stoname |  Set-AzStorageBlobInventoryPolicy -Policy (@{
```

Comment:
Should we also update the test case for input from json

Reply:
Yes for sure. Updated the case.